cunumeric/module.py

```diff
@@ -1748,15 +1761,22 @@ def concatenate(
     --------
     Multiple GPUs, Multiple CPUs
     """
+    if dtype is not None and out is not None:
```

Comment:
@bryevdv Please help to confirm if we really need to check this here for the same behavior with Numpy.

Reply:
It seems to be:
```python
In [2]: import numpy as np

In [3]: a = np.array([1, 2, 3, 4])

In [4]: b = np.array([1, 2, 3, 4])

In [5]: c = np.empty(8, dtype=np.int16)

In [6]: np.concatenate([a, b], out=c, dtype=np.float32)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 1
----> 1 np.concatenate([a, b], out=c, dtype=np.float32)

File <__array_function__ internals>:200, in concatenate(*args, **kwargs)

TypeError: concatenate() only takes `out` or `dtype` as an argument, but both were provided.
```

Reply:
Then we will report TypeError in cuNumeric, too.